llvm/lib/MC/GOFFObjectWriter.cpp

```diff
@@ -239,6 +298,63 @@ class GOFFWriter {
 GOFFWriter::GOFFWriter(raw_pwrite_stream &OS, MCAssembler &Asm)
     : OS(OS), Asm(Asm) {}
 
+void GOFFWriter::defineSectionSymbols(const MCSectionGOFF &Section) {
+  if (Section.isSD()) {
+    GOFFSymbol SD(Section.getName(), Section.getId(),
+                  Section.getSDAttributes());
+    writeSymbol(SD);
+  }
+
+  if (Section.isED()) {
+    GOFFSymbol ED(Section.getName(), Section.getId(),
+                  Section.getParent()->getId(), Section.getEDAttributes());
+    if (Section.requiresLength())
```

Comment:
Do we really need a special flag for this?   As I understand it, the length should remain zero for EDs that have PRs inside, i.e. those that use GOFF::ESD_BA_Merge - can't we just check for that?   Also, in those cases no text should ever get emitted into the ED section, so it's size should be zero naturally?

Reply:
> so it's size should be zero naturally?

Yes, that's true, and it makes the code easier.